src/lib/apiTypesV2/Subscription.cpp

```diff
@@ -120,12 +120,34 @@ std::string Condition::toJson()
 
   JsonHelper jhe;
 
-  jhe.addString("q", this->expression.q);
-  jhe.addString("geometry", this->expression.geometry);
-  jhe.addString("coords", this->expression.coords);
-  jhe.addString("georel", this->expression.georel);
-  
-  jh.addRaw("expression", jhe.str());
+  bool atLeastOneSubField = false;
+
+  if (this->expression.q != "")
+  {
+    atLeastOneSubField = true;
+    jhe.addString("q", this->expression.q);
+  }
+  if (this->expression.geometry != "")
+  {
+    atLeastOneSubField = true;
+    jhe.addString("geometry", this->expression.geometry);
+  }
+  if (this->expression.coords != "")
+  {
+    atLeastOneSubField = true;
+    jhe.addString("coords", this->expression.coords);
+  }
+  if (this->expression.georel != "")
+  {
+    atLeastOneSubField = true;
+    jhe.addString("georel", this->expression.georel);
+  }
+
+  if (atLeastOneSubField)
+  {
+    atLeastOneSubField = true;
```

Comment:
This last 'atLeastOneSubField = true' doesn't seem necessary ...
Also, you could remove them all and replace with a **long** if statement:

    if ((expression.q != "") || (expression.geometry != "") || ...)
    {
      jh.addRaw("expression", jhe.str());
    }

Or better, perhaps:    if (jhe.str() != "") ?